### PR TITLE
[v3] Hardcoding decimals in chain config to reduce complexity

### PR DESCRIPTION
**Pull request by ekenigs**
### Description

Hardcoding decimals in chain config to reduce complexity
* Hardcoded decimals and removed complexity of getting decimals
* Renamed `assetsData` to `assets` and created a new class `ChainAsset` and moved it to `Chain` class
* Added `nativeAsset` to `Chain`

### Checklist

- [ ] If this requires a documentation change, I have created a PR in [moonbeam-docs](https://github.com/moonbeam-foundation/moonbeam-docs) repository.
- [ ] If this requires it, I have updated the Readme
- [ ] If necessary, I have updated the examples
- [ ] I have verified if I need to create/update unit tests
- [ ] I have verified if I need to create/update acceptance tests
- [ ] If necessary, I have run acceptance tests on this branch in CI

### CODE REVIEW

File: packages/sdk/src/polkadot/PolkadotService.ts [hunk 1]
```diff
@@ -4,84 +4,57 @@ import {
...
       throw new Error('No native symbol key found');
     }
 
-    const asset = this.configService.getAsset(key);
-
-    if (!asset) {
-      throw new Error(`No asset found for key "${key}" and symbol "${symbol}"`);
+    // TODO: Remove this once Equilibrium is updated
+    // or find better way if issue appears on other chains
+    if (key === 'token' && [equilibrium.key].includes(this.chain.key)) {
```
Comment (Rihyx) [inline packages/sdk/src/polkadot/PolkadotService.ts:53]: I don't understand why you need this if you do the same `return this.chain.getChainAsset(key);` below.

**Review by Rihyx — COMMENTED** [review]
Wrote some comments but I will later check again and test

File: packages/config/src/types/AssetConfig.ts [hunk 2]
```diff
@@ -25,7 +25,8 @@ export interface DestinationFeeConfig extends FeeAssetConfig {
 export interface FeeAssetConfig {
   asset: Asset;
   balance: BalanceConfigBuilder;
-  xcmDeliveryFeeAmount?: number;
+  // Sometimes we need to add some extra amount to a fee that is returned by "paymentInfo" for extrinsic to not fail.
+  extra?: number;
 }
 
 export class AssetConfig {
```
[the remark this answers is not in the document]
Comment (ekenigs) [reply packages/config/src/types/AssetConfig.ts:32]: I will do it in the next PR.

File: packages/config/src/chains.ts [hunk 3]
```diff
@@ -1456,66 +1663,60 @@ export const peaqEvmAlphanet = new EvmParachain({
...
+    }),
+    ChainAsset.fromAsset(usdcwh, {
       address: '0xFffFffFF000000000000000000000000000003E9',
-      asset: usdcwh,
-      id: '0xFffFffFF000000000000000000000000000003E9',
-      metadataId: 1001,
-      minId: 1001,
-    },
-    {
+      decimals: 0, // todo
```
[the remark this answers is not in the document]
Comment (ekenigs) [reply packages/config/src/chains.ts:1687]: Somehow missed in a few places. But I filled them now.

File: packages/config/src/chains.ts [hunk 4]
```diff
@@ -82,713 +83,819 @@ import {
...
+      ids: {
+        id: { VToken: 'KSM' },
+      },
+    }),
+    ChainAsset.fromAsset(vmovr, {
+      decimals: 18,
+      ids: {
+        id: { VToken: 'MOVR' },
+      },
+    }),
```
[the remark this answers is not in the document]
Comment (ekenigs) [reply packages/config/src/chains.ts]: Changed.

File: packages/sdk/src/polkadot/PolkadotService.ts (hunk 1 above)
Comment (ekenigs) [reply packages/sdk/src/polkadot/PolkadotService.ts:53]: I think I will remove the whole method.

File: packages/config/src/chains.ts [hunk 5]
```diff
@@ -847,235 +957,282 @@ export const moonbaseBeta = new EvmParachain({
...
+        id: '238111524681612888331172110363070489924',
+      },
+    }),
+    ChainAsset.fromAsset(para, {
       address: '0xFfFffFFF18898CB5Fe1E88E668152B4f4052A947',
-      asset: para,
-      id: '32615670524745285411807346420584982855',
-    },
-    {
+      decimals: 18,
```
Comment (mmaurello) [inline packages/config/src/chains.ts:1103]:
```suggestion
      decimals: 12,
```

**Review by mmaurello — CHANGES_REQUESTED** [review]

File: packages/config/src/chains.ts (hunk 5 above)
Comment (ekenigs) [reply packages/config/src/chains.ts:1103]: Good catch!

**Review by mmaurello — APPROVED** [review]

**Pull request merged**